memory/index.js

```diff
@@ -14,34 +14,21 @@
  * limitations under the License.
  */
 
-let unsupported;
-if (typeof navigator !== 'undefined' && 'deviceMemory' in navigator) {
-  unsupported = false;
-} else {
-  unsupported = true;
-}
-let memoryStatus;
-if (!unsupported) {
-  const performanceMemory = 'memory' in performance ? performance.memory : null;
-  memoryStatus = {
-    unsupported,
+const useMemoryStatus = (initialMemoryStatus = null) => {
```

Comment:
@wingleung 

Do you think we could move `supported` and `performanceMemory` out of the hook?

Reply:
does `performanceMemory` need to be out of the hook? I think we need it at hook execution because we want to see the state of the performance memory at the time the hook gets called. For example for `performanceMemory.usedJSHeapSize`